modules/visualisation/src/backend/api.ts

```diff
@@ -0,0 +1,78 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import ConfusionMatrix from 'ml-confusion-matrix'
+import { Matrix } from 'ml-matrix'
+import nanoid from 'nanoid'
+
+import { getTrainTestDatas } from '../tools/data_loader'
+import {
+  computeConfusionMatrix,
+  computeEmbeddingSimilarity,
+  computeIntentSimilarity,
+  computeOutliers,
+  computeScatterEmbeddings
+  // computeTsneScatterEmbeddings
+} from '../tools/visualisation'
+
+import { VisuState } from './typings'
+
+export default async (bp: typeof sdk, state: VisuState) => {
+  const longJobsPool = {}
+  const router = bp.http.createRouterForBot('new_qna')
+  const glob_res = []
+
+  router.get('/confusionMatrix', async (req, res) => {
```

Comment:
Please add the async middleware so the server doesn't crash if an error is thrown by one of your method